datasync/downloader/downloader.go

```diff
@@ -1567,6 +1605,14 @@ func (d *Downloader) commitFastSyncData(results []*fetchResult, stateSync *state
 	for i, result := range results {
 		blocks[i] = types.NewBlockWithHeader(result.Header).WithBody(result.Transactions)
 		receipts[i] = result.Receipts
+		if result.StakingInfo != nil {
```

Comment:
Isn't `AddStakingInfoToDB` necessary process? If this nil checking logic doesn't exit, `AddStakingInfoToDB` will return an error for `StakingInfo == nil` case. But, this if branch miss the error handling

Reply:
It seems that `stakingManager` exists only for `WeightedRandom` proposer policy. Is it okay that the info is missing when the proposer policy is used

Reply:
I am going to resolve this with another PR.